File: 2444-count-subarrays-with-fixed-bounds/2444-count-subarrays-with-fixed-bounds.java
class Solution {
    public long countSubarrays(int[] nums, int minK, int maxK) {
        if(minK>maxK)return 0;
        int n = nums.length;
        long res = 0;
        int mn=-1,mx=-1,l=-1;
        for(int i=0;i<n;i++){
           if(nums[i]<minK||nums[i]>maxK)l=i;
           if(nums[i]==minK)mn=i;
           if(nums[i]==maxK)mx=i;
           int r = Math.min(mn,mx);
           if(r>l)res+=r-l;
        }
        return res;
    }

    
}

File: 2444-count-subarrays-with-fixed-bounds/2444-count-subarrays-with-fixed-bounds.py
class Solution:
    def countSubarrays(self, nums: List[int], minK: int, maxK: int) -> int:
        answer = 0
        invalid = -1
        mnVal = -1
        mxVal = -1

        for i in range(len(nums)):
            if nums[i] < minK or nums[i] > maxK:
                invalid = i
                continue
            mnVal = i if nums[i] == minK else mnVal
            mxVal = i if nums[i] == maxK else mxVal

            if invalid < mnVal and invalid < mxVal:
                answer += min(mnVal, mxVal) - invalid

        return answer